### PR TITLE
Update to CBMC 5.37

### PR DESCRIPTION
**Pull request by FrNecas**
Related: https://github.com/peterschrammel/cbmc/pull/26

I nitially wanted to take a few more steps in between but I had to skip basically straight to 5.22 due to Glucose not being compilable in CBMC up until that point and there were not that many changes, each release introduced a few at most. Some of these changes are quite large, e.g. workaround for `exprt.opX()` not being available anymore.

### CODE REVIEW

**Review by peterschrammel — APPROVED** [review]

File: src/domains/util.cpp [hunk 1]
```diff
@@ -38,34 +38,36 @@ void extend_expr_types(exprt &expr)
     return;
   if(expr.id()==ID_unary_minus)
   {
-    extend_expr_types(expr.op0());
+    exprt &op0=to_unary_minus_expr(expr).op();
```
Comment (viktormalik) [inline src/domains/util.cpp:42]: `op0` is a weird variable name since we have an unary operator here. Better use `op` instead.

**Review by viktormalik — COMMENTED** [review]
One nit, looks good otherwise.

File: src/domains/util.cpp (hunk 1 above)
Comment (FrNecas) [reply src/domains/util.cpp]: Fixed also a bit lower in the method with typecast expression (also unary). Rebased and squashed.

**Review by viktormalik — APPROVED** [review]

**Comment by viktormalik** [comment]
I believe that this closes #98.

**Pull request merged**